PhysicsTools/PatAlgos/plugins/LeptonUpdater.cc

```diff
@@ -181,9 +181,15 @@ void pat::LeptonUpdater<T>::produce(edm::StreamID, edm::Event &iEvent, edm::Even
       float signPV = 1.;
       float signBS = 1.;
       if (beamSpotIsValid) {
-        signBS = copysign(1., lep.bestTrack()->dxy(beamSpot));
+        if constexpr (std::is_same_v<T, pat::Muon>)
```

Comment:
I would rather consider the `pat::electron` as the "special case", to be considered separately from "all other cases" (ok, by now "all other cases" is just `pat::muon` indeed).
Possible suggestion:
```C++
        if constexpr (std::is_same_v<T, pat::Electron>)
          signBS = copysign(1., lep.gsfTrack()->dxy(beamSpot));
        else
          signBS = copysign(1., lep.bestTrack()->dxy(beamSpot));
```
(and the same for `signPV` here below)

Reply:
Thanks, good point